pkg/executor/infoschema_reader.go

```diff
@@ -706,10 +716,82 @@ func (e *memtableRetriever) setDataFromTables(ctx context.Context, sctx sessionc
 		return nil
 	}
 
+	// Special optimize for queries on infoschema v2 like:
+	//     select count(table_schema) from INFORMATION_SCHEMA.TABLES
+	//     select count(*) from INFORMATION_SCHEMA.TABLES
+	//     select table_schema, table_name from INFORMATION_SCHEMA.TABLES
+	// column pruning in general is not supported here.
+	if onlySchemaOrTableColumns(e.columns) {
+		is := e.is
+		if raw, ok := is.(*infoschema.SessionExtendedInfoSchema); ok {
+			is = raw.InfoSchema
+		}
+		v2, ok := is.(interface {
+			IterateAllTableItems(visit func(infoschema.TableItem) bool)
+		})
+		if ok {
+			if x := ctx.Value("cover-check"); x != nil {
```

Comment:
maybe use a failpoint

Reply:
IMO, failpoint is not as convient sometimes. 
- It works globally, so there's no good way for precise control with a specific query.
- It requires `make failpoint-enable`, an extra step to enable the test

Reply:
> It requires make failpoint-enable, an extra step to enable the test

I have added this feature for failpoint, it can be configurated with our other testing command line arguments like `--tags=intest` or `-race` https://github.com/pingcap/failpoint?tab=readme-ov-file#quick-start-use-failpoint-toolexec